test/test_index.py

```diff
@@ -44,6 +44,31 @@ def test_bare(self):
         index = self.repo.index
         self.assertEqual(len(index), 0)
 
+    def test_conflicts_in_bare_repository(self):
+        def create_conflict_file(repo, branch, content):
+            oid = repo.create_blob(content)
+            tb = repo.TreeBuilder()
+            tb.insert('conflict', oid, pygit2.GIT_FILEMODE_BLOB)
+            tree = tb.write()
+
+            sig = pygit2.Signature('Author', 'author@example.com')
+            commit = repo.create_commit(branch.name, sig, sig,
+                    'Conflict', tree, [branch.target])
+            self.assertIsNotNone(commit)
+            return commit
+
+        b1 = self.repo.create_branch('b1', self.repo.head.peel())
+        c1 = create_conflict_file(self.repo, b1, 'Conflict 1')
+        b2 = self.repo.create_branch('b2', self.repo.head.peel())
+        c2 = create_conflict_file(self.repo, b2, 'Conflict 2')
+
+        index = self.repo.merge_commits(c1, c2)
+        self.assertIsNotNone(index.conflicts)
+
+        for (a, t, o) in index.conflicts:
+            diff = index.diff_index_entries(a, t, o)
+            self.assertIsNotNone(diff)
+            self.assertGreater(len(diff), 0)
```

Comment:
Since we know what should come out, I think it'd be better if we compared against a string with the exact expected output, rather than just making sure it's not an empty string.